pkg/certs/cert-inspection/certgraphanalysis/disk_certs.go

```diff
@@ -0,0 +1,180 @@
+package certgraphanalysis
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/openshift/library-go/pkg/certs/cert-inspection/certgraphapi"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/cert"
+)
+
+func gatherSecretsFromDisk(ctx context.Context, prefix, dir string, options certGenerationOptionList) ([]*certgraphapi.CertKeyPair, error) {
+	ret := []*certgraphapi.CertKeyPair{}
+	parentDir := filepath.Join(prefix, dir)
+	_, err := os.Stat(parentDir)
+	if os.IsNotExist(err) {
+		return ret, nil
+	}
+
+	fmt.Fprintf(os.Stdout, "Gathering secrets from %s.\n", parentDir)
+	err = filepath.WalkDir(parentDir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() {
+			return nil
+		}
+
+		fmt.Fprintf(os.Stdout, "Checking if %s is a secret.\n", path)
+		if detail, err := parseFileAsSecretKey(path, prefix, dir); err == nil && detail != nil {
+			fmt.Fprintf(os.Stdout, "Found secret key in %s.\n", path)
+			options.rewriteCertKeyPair(metav1.ObjectMeta{}, detail)
+			ret = append(ret, detail)
+			return nil
+		}
+
+		if detail, err := parseFileAsCertificate(path, prefix, dir); err == nil && detail != nil {
+			fmt.Fprintf(os.Stdout, "Found certificate in %s.\n", path)
+			options.rewriteCertKeyPair(metav1.ObjectMeta{}, detail)
+			ret = append(ret, detail)
+			return nil
+		}
+
+		return nil
+	})
+	return ret, err
+}
+
+func parseFileAsSecretKey(path, prefix, dir string) (*certgraphapi.CertKeyPair, error) {
+	bytes, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	block, _ := pem.Decode(bytes)
+	if block == nil || block.Type != "RSA PRIVATE KEY" {
+		return nil, fmt.Errorf("PEM block type must be RSA PRIVATE KEY")
+	}
+	key, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	pathWithoutPrefix := strings.Replace(path, prefix, "", 1)
+	return &certgraphapi.CertKeyPair{
+		Spec: certgraphapi.CertKeyPairSpec{
+			SecretLocations: nil,
+			OnDiskLocations: []certgraphapi.OnDiskCertKeyPairLocation{
+				{
+					Key: certgraphapi.OnDiskLocation{
+						Path: pathWithoutPrefix,
+						// TODO[vrutkovs]: fill in other settings
```

Comment:
permissions and se linux labels as likely next steps.  Followup fine.

Reply:
> permissions and se linux labels as likely next steps. Followup fine.

A peer to https://github.com/openshift/library-go/blob/master/pkg/certs/cert-inspection/certgraphapi/types.go#L15 would allow a spot to store things like the the file metadata in a spot peer to the inresource names and annotations to allow similar consuming code to be written.